### PR TITLE
Guestbook

### PR DESCRIPTION
**Pull request by DCamaron1**

### CODE REVIEW

File: src/guestbook/Guest.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package guestbook;
+
+interface Guest{
+	String getGuestListing();
+}
```
Comment (epuzio) [inline src/guestbook/Guest.java:5]: remove 🚫 ◀️

File: src/guestbook/Guestbook.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+
+public class Guestbook implements ActionListener {
+	JPanel panel = new JPanel();
+	JFrame frame = new JFrame();
+	JTextField input = new JTextField(20);
+	JButton showList = new JButton();
+	JButton addVIPButton = new JButton();
+	JButton addRegularButton = new JButton();
```
Comment (epuzio) [inline src/guestbook/Guestbook.java]: good variable names!

File: src/guestbook/Guestbook.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	ArrayList<Guest> list = new ArrayList<>();
+
+	public void CreateUI() {
+		frame.add(panel);
+		frame.setVisible(true);
+		frame.setSize(300, 100);
+		frame.setTitle("Hotel Java Guestbook");
```
Comment (epuzio) [inline src/guestbook/Guestbook.java:27]: oh that's cool, nice title change

File: src/guestbook/Guestbook.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	public String prettyList() {
+		String fullList = "";
+		for (int i = 0; i < list.size(); i++) {
+			fullList = fullList + list.get(i) + "\n";
+		}
+		return fullList;
+
+	}
```
Comment (epuzio) [inline src/guestbook/Guestbook.java:53]: that's a really good way to make a nice-looking list!

File: src/guestbook/Guestbook.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+	@Override
```
Comment (epuzio) [inline src/guestbook/Guestbook.java]: delete @ Override 🚗

File: src/guestbook/Guestbook.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	JButton addVIPButton = new JButton();
+	JButton addRegularButton = new JButton();
+	ArrayList<Guest> list = new ArrayList<>();
+
+	public void CreateUI() {
+		frame.add(panel);
+		frame.setVisible(true);
+		frame.setSize(300, 100);
+		frame.setTitle("Hotel Java Guestbook");
+		panel.add(input);
```
Comment (IsisJ) [inline src/guestbook/Guestbook.java:28]: Oh! You gave your frame a name 👍

File: src/guestbook/Guestbook.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+		if (buttonPressed == addVIPButton) {
+			String name = input.getText();
+			list.add(new VIP(name));
+		}
+
+		if (buttonPressed == addRegularButton) {
+			String name = input.getText();
+			list.add(new Regular(name));
+
```
Comment (epuzio) [inline src/guestbook/Guestbook.java]: unnecessary space

File: src/guestbook/VIP.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
+package guestbook;
+
+public class VIP implements Guest{
+	String name; 
+	
+	public VIP(String name) {
+		this.name= "VIP: " + name;
+	}
+	
+	@Override
```
Comment (epuzio) [inline src/guestbook/VIP.java]: delete @ Override 🏭

File: src/guestbook/Regular.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
+package guestbook;
+
+public class Regular implements Guest{
+	String name;
+	
+	public Regular(String name) {
+		this.name=name;
+	}
+	
+	@Override
```
Comment (epuzio) [inline src/guestbook/Regular.java]: delete @ Override 👺

File: src/guestbook/Guestbook.java [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+		user.CreateUI();
+	}
+
+	public String prettyList() {
+		String fullList = "";
+		for (int i = 0; i < list.size(); i++) {
+			fullList = fullList + list.get(i) + "\n";
+		}
+		return fullList;
+
```
Comment (epuzio) [inline src/guestbook/Guestbook.java]: messy formatting, try ctrl+f

File: src/guestbook/Guestbook.java [hunk 11]
```diff
@@ -0,0 +1,74 @@
+package guestbook;
+
+import java.awt.Color;
```
Comment (epuzio) [inline src/guestbook/Guestbook.java:3]: why import java.awt.Color?

File: src/guestbook/Guestbook.java [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+import java.util.ArrayList;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+
+public class Guestbook implements ActionListener {
+	JPanel panel = new JPanel();
```
Comment (mwoguri) [inline src/guestbook/Guestbook.java]: Instance variables can be private.

File: src/guestbook/Guestbook.java [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+	JFrame frame = new JFrame();
+	JTextField input = new JTextField(20);
+	JButton showList = new JButton();
+	JButton addVIPButton = new JButton();
+	JButton addRegularButton = new JButton();
+	ArrayList<Guest> list = new ArrayList<>();
+
+	public void CreateUI() {
```
Comment (mwoguri) [inline src/guestbook/Guestbook.java]: methods should start with lowercase letters.

File: src/guestbook/Guestbook.java (hunk 3 above)
Comment (mwoguri) [inline src/guestbook/Guestbook.java:27]: Cute title!

File: src/guestbook/VIP.java [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+	public String getGuestListing() {
+		
+		return null;
+	}
+
```
Comment (IsisJ) [inline src/guestbook/VIP.java:13]: Do you ever use this method ❓

File: src/guestbook/Regular.java [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+	public String getGuestListing() {
+		return null;
```
Comment (mwoguri) [inline src/guestbook/Regular.java]: Why does this return null?

File: src/guestbook/VIP.java [hunk 16]
```diff
@@ -0,0 +1,21 @@
+package guestbook;
+
+public class VIP implements Guest{
+	String name; 
+	
+	public VIP(String name) {
+		this.name= "VIP: " + name;
```
Comment (mwoguri) [inline src/guestbook/VIP.java]: Please just store the name in the `name` instance variable.

File: src/guestbook/VIP.java [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+	}
+	
+	@Override
+	public String getGuestListing() {
+		
+		return null;
```
Comment (mwoguri) [inline src/guestbook/VIP.java]: returns null?  That doesn't seem right...

File: src/guestbook/Guestbook.java [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+
+	public static void main(String[] args) {
+		Guestbook user = new Guestbook();
+		user.CreateUI();
+	}
+
+	public String prettyList() {
+		String fullList = "";
+		for (int i = 0; i < list.size(); i++) {
+			fullList = fullList + list.get(i) + "\n";
```
Comment (mwoguri) [inline src/guestbook/Guestbook.java]: You're making use of the `toString()` method, which works in this case...but could you try using the method that the interface provides?

**Review by mwoguri — APPROVED** [review]
LGTM!

**Pull request merged**